packages/@aws-cdk/aws-cloudwatch/lib/query.ts

```diff
@@ -0,0 +1,75 @@
+import * as cdk from '@aws-cdk/core';
+import { ConcreteWidget } from "./widget";
+
+/**
+ * Properties for a Query widget
+ */
+export interface QueryWidgetProps {
+  /**
+   * Title for the graph
+   *
+   * @default No title
+   */
+  readonly title?: string;
+
+  /**
+   * Log group to query
+   */
+  readonly logGroup: string;
```

Comment:
Feels like this should accept an `ILogGroup`.

Reply:
Ok I'll check that out, I'm pretty new with CDK

Reply:
This would add a cyclic dependency:

```
lerna ERR! ECYCLE @aws-cdk/aws-cloudwatch -> @aws-cdk/aws-logs -> @aws-cdk/aws-cloudwatch
```

So I'm think that renaming it to `logGroupName` and keeping it a string would be the best course of action

Reply:
I'd really really like users to be able to pass the object though. We will have to follow the existing dependency arrows while coming up with our solution. I'm thinking something like this:

```ts
// cloudwatch
interface IQueryLogGroup {
  public readonly logGroupName: string;
}

// logs
interface ILogGroup extends cloudwatch.IQueryLogGroup {
 // ...
}
```

Reply:
Ok so I rebased and added an `IQueryLogGroup`.

Is the `ILogGroup extends cloudwatch.IQueryLogGroup` required? For TypeScript an `ILogGroup` can be supplied where an `IQueryLogGroup` is required without this inheritance (it is sort of an ugly dependency - I don't know if such a thing already exists in CDK).

Perhaps it is required for other languages using JSII etc? It might be a fair trade off to not add this inheritance (until the full solution is decided). I'm thinking that would only impact users who are using more type strict languages like Java but even so an anonymous class could still work.

The other option is to make `QueryWidgetProps#logGroup` take an `IQueryLogGroup` or a `string` and at least then with TypeScript you can pass the `ILogGroup` and in other languages you still have the option to pass the name directly as a `string`?

Let me know which you prefer and I will make the change.

Reply:
> Perhaps it is required for other languages using JSII etc?

Exactly. This class could not be used in typed languages that use nominal subtyping (Java, C#), which would be a shame. Treating them as second class citizens (by forcing them to custom-implement an anonymous subclass) is not really acceptable in my opinion. It's not very discoverable either, and would present a big hurdle to adoption.

I agree with you that this solution is not fantastic, but it's at least pragmatic and works well enough until we have better alternatives.

> The other option is to make `QueryWidgetProps#logGroup` take an `IQueryLogGroup` or a `string`

You are thinking a union type here, which we as a rule don't use in L2 constructs because they won't translate well to Go.

I can understand you feeling dissatisfied with the available options. If it helps, I will take all of the design blame :).

Reply:
I mean I don't personally mind; you have the bigger picture here. So will I go with what you originally proposed?

Reply:
Yes please

Reply:
Done and rebased

(sorry about the noise below; I'm having issues building the latest version for some reason)